ddtrace/contrib/kafka/patch.py

```diff
@@ -230,10 +221,12 @@ def traced_commit(func, instance, args, kwargs):
         return func(*args, **kwargs)
 
     if config._data_streams_enabled:
-        message = get_argument_value(args, kwargs, 0, "message")
-        offsets = kwargs.get("offsets", [])
+        message = get_argument_value(args, kwargs, 0, "message", True)
+        # messgae and offset are mutally exclusive. Only one parameter can be passed.
         if message is not None:
             offsets = [TopicPartition(message.topic(), message.partition(), offset=message.offset())]
+        else:
+            offsets = get_argument_value(args, kwargs, 1, "offsets", True)
```

Comment:
If this returns `None`, then don't we try to iterate through `None` which will throw an error?

Reply:
Consumer.commit() can be called with the following arguments: `message`, `offsets` and `asynchronous`. 

The following function calls are valid:
`Consumer.commit(message=....)`
`Consumer.commit(message=..., asynchronous=....)`
`Consumer.commit(offsets=None, asynchronous=...)`
`Consumer.commit(asynchronous=...)`

The following function calls are INVALID. This raises a ValueError:
`Consumer.commit(message=...., offsets=.....)`


There is a scenario where both message and offsets are not passed as keyword arguments. In this case `get_argument_value(....)` should not raise an ArgumentError. It should just return `None`.



Docs for reference: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.commit